TST: test_data_name_third_party_package fails when OS is case-insensitive

I noticed this when I ran remote data tests on Windows and my path had mixed case. Not sure if it is worth fixing. 🤷 
```
_____________________ test_data_name_third_party_package ______________________

    def test_data_name_third_party_package():
        """Regression test for issue #1256

        Tests that `get_pkg_data_filename` works in a third-party package that
        doesn't make any relative imports from the module it's used from.

        Uses a test package under ``data/test_package``.
        """

        # Get the actual data dir:
        data_dir = os.path.join(os.path.dirname(__file__), "data")

        sys.path.insert(0, data_dir)
        try:
            import test_package

            filename = test_package.get_data_filename()
>           assert filename == os.path.join(data_dir, "test_package", "data", "foo.txt")
E           AssertionError: assert 'D:\\...\\S...data\\foo.txt' == 'd:\\...\\st...data\\foo.txt'
E             - D:\...\STScI\ssb\astropy\astropy\utils\tests\data\test_package\data\foo.txt
E             ? ^      ^^^ ^
E             + d:\...\stsci\ssb\astropy\astropy\utils\tests\data\test_package\data\foo.txt
E             ? ^      ^^^ ^

astropy\utils\tests\test_data.py:888: AssertionError
```

--- a/astropy/utils/tests/test_data.py
+++ b/astropy/utils/tests/test_data.py
@@ -885,7 +885,9 @@ def test_data_name_third_party_package():
         import test_package
 
         filename = test_package.get_data_filename()
-        assert filename == os.path.join(data_dir, "test_package", "data", "foo.txt")
+        assert os.path.normcase(filename) == (
+            os.path.normcase(os.path.join(data_dir, "test_package", "data", "foo.txt"))
+        )
     finally:
         sys.path.pop(0)
 

